### PR TITLE
Done pucchi :*

### PR DESCRIPTION
**Pull request by chintanb**

### CODE REVIEW

File: Crypto-ass1/PrimeSieve.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
+mport java.util.*;
```
Comment (chintanb) [inline Crypto-ass1/PrimeSieve.java]: Do change with proper Java-Doc and better print statements.
Also focus on the formatting too ;)

File: Crypto-ass1/PrimeSieve.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+	public static void main(String[] args) {
+		int N = 9999;
+
+        boolean[] isPrime = new boolean[N + 1];
+        List<Integer> desiredPrimes = new ArrayList<>();
+        for (int i = 0; i <= N; i++) {
+            isPrime[i] = true;
+        }
+
```
Comment (chintanb) [inline Crypto-ass1/PrimeSieve.java]: ^ Putting all the numbers as prime, initially.

File: Crypto-ass1/PrimeSieve.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            if (isPrime[i]) {
+                for (int j = i*2 ; j <= N; j+=i) {
+                    isPrime[j] = false;
+                }
+            }
+        }
+
+        // count primes
+        int primes = 0;
+        for (int i = 1000; i <= N; i++) {
```
Comment (chintanb) [inline Crypto-ass1/PrimeSieve.java]: Putting in to a separate list with primes starting from 1000 ;)
Saving one for loop, hence improved the complexity.

File: Crypto-ass1/PrimeSieve.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            if (isPrime[i]) {
+                desiredPrimes.add(i);
+                System.out.println(i);
+                primes++;
+            }
+        }
+        
+        System.out.println("The number of primes <= " + N + " is " + primes);
```
Comment (chintanb) [inline Crypto-ass1/PrimeSieve.java]: Printing total count of primes.

File: Crypto-ass1/PrimeSieve.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        
+        List<Integer> firstHudreds = desiredPrimes.subList(0, 99);
```
Comment (chintanb) [inline Crypto-ass1/PrimeSieve.java]: Creating two sub-lists with first 100 and last 100 primes.
Its better we check the total size first. Gives a better program, less exception (ArrayIndexOutOfBound) prone.

File: Crypto-ass1/PrimeSieve.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        
+        List<Integer> lastHudreds = desiredPrimes.subList(
+                                            desiredPrimes.size()-99, 
+                                            desiredPrimes.size());
+        
+        Integer multiple = 1;
+        for(Integer prime : firstHudreds) {
+            System.out.println(prime);
+            multiple *= prime;
```
Comment (chintanb) [inline Crypto-ass1/PrimeSieve.java]: Counting the multiple of each element of `firstHundreds` ;)

File: Crypto-ass1/PrimeSieve.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        }
+            
+        for(Integer prime : lastHudreds) {
+            System.out.println(prime);
+        }                             
+        
+        System.out.println("Multiple of first 100 primes is," + multiple);
```
Comment (chintanb) [inline Crypto-ass1/PrimeSieve.java]: And printing the multiple, and we lived happily ever after.

**Comment by chintanb** [comment]
@Radhika12 Please review :*

**Comment by chintanb** [comment]
> Please don't judge me based on this code. This can be much more improved.
Ultimately, this is not the best, but this is what I could do at freaking 12 am.

**Comment by Radhika12** [comment]
Love you puchii :* :D

**Pull request merged**